src/features/DisplayControls/NursingTasks/components/TaskTile.jsx

```diff
@@ -54,7 +54,38 @@ export default function TaskTile(props) {
     );
   };
 
-  const iconType = isLateTask() ? "Late" : "Pending";
+  const iconType = () => {
+    if (newMedicationNursingTask.administeredTimeInEpochSeconds) {
+      const administeredLateWindowInSeconds =
+        startTimeInEpochSeconds +
+        nursingTasks.timeInMinutesFromStartTimeToShowAdministeredTaskAsLate *
+          60;
+      const administeredTimeInSeconds =
+        newMedicationNursingTask.administeredTimeInEpochSeconds / 1000;
+      const isLate =
+        administeredTimeInSeconds > administeredLateWindowInSeconds;
+
+      return isLate ? "Administered-Late" : "Administered";
+    }
+
+    return isLateTask() ? "Late" : "Pending";
+  };
+
+  const getTime = () => {
```

Comment:
can we please move these helper methods to TaskTile or Nursing Tasks utils file?